tests/integration_tests/functional/test_drives.py

```diff
@@ -57,6 +58,54 @@ def test_rescan(test_microvm_with_ssh, network_config):
     )
 
 
+def test_rescan_dev(test_microvm_with_ssh, network_config):
+    """Verify that rescan works with a device-backed virtio device."""
+    test_microvm = test_microvm_with_ssh
+    # To access a block device, seccomp_level = 2 is too restrictive
+    test_microvm.jailer.seccomp_level = 1
+    test_microvm.spawn()
+
+    # Set up the microVM with 1 vCPUs, 256 MiB of RAM, 0 network ifaces and
+    # a root file system with the rw permission. The network interface is
+    # added after we get a unique MAC and IP.
+    test_microvm.basic_config()
+
+    _tap, _, _ = test_microvm_with_ssh.ssh_network_config(network_config, '1')
+
+    # Add a scratch block device.
+    fs1 = drive_tools.FilesystemFile(os.path.join(test_microvm.fsfiles, 'fs1'))
+    response = test_microvm.drive.put(
+        drive_id='scratch',
+        path_on_host=test_microvm.create_jailed_resource(fs1.path),
+        is_root_device=False,
+        is_read_only=False
+    )
+    assert test_microvm.api_session.is_status_no_content(response.status_code)
+
+    test_microvm.start()
+
+    ssh_connection = net_tools.SSHConnection(test_microvm.ssh_config)
+
+    _check_scratch_size(ssh_connection, fs1.size())
+
+    fs2 = drive_tools.FilesystemFile(
+        os.path.join(test_microvm.fsfiles, 'fs2'),
+        size=512
+    )
+
+    losetup = ['losetup', '--find', '--show', fs2.path]
+    losetup_result = run(losetup, stdout=PIPE, check=True)
+    loopback_device = losetup_result.stdout.decode('utf-8').rstrip()
+
+    response = test_microvm.drive.patch(
+        drive_id='scratch',
+        path_on_host=test_microvm.create_jailed_resource(loopback_device),
+    )
+    assert test_microvm.api_session.is_status_no_content(response.status_code)
+
+    _check_scratch_size(ssh_connection, fs2.size())
+
```

Comment:
Cleanup for the created loop device, from host.

```suggestion
    lodetach = ['losetup', '-d', loopback_device]
    run(lodetach, stdout=PIPE, check=True)
```